src/chain/elders_info.rs

```diff
@@ -80,25 +80,32 @@ impl EldersInfo {
 
     /// Creates a new `EldersInfo` by merging this and the other one.
     pub fn merge(&self, other: &Self) -> Result<Self, RoutingError> {
-        let members = self.members.iter().chain(&other.members).cloned().collect();
+        let members = self
+            .members
+            .iter()
+            .chain(&other.members)
+            .map(|(pub_id, p2p_node)| (*pub_id, p2p_node.clone()))
+            .collect();
         Self::new(members, self.prefix.popped(), vec![self, other])
     }
 
     // WIP: remove me (or make less heavy)
     pub fn members(&self) -> BTreeSet<PublicId> {
```

Comment:
Probably this is problematic, at call site it looks cheap to call, it probably should not look like a straight accessor.
Maybe best to rename to something like `members_as_set`.

Also should it better be an iterator that we return? (it depends how often it is used and how).

Reply:
>   Maybe best to rename to something like members_as_set.
Agree
> Also should it better be an iterator that we return? (it depends how often it is used and how).
The reason we return sets is mostly because we call `contains` on them. However yeah we can probably call `contains_key` on the underlying map directly instead. Maybe even get rid of the method all together